prime-router/src/main/kotlin/cli/helpers/HL7DiffHelper.kt

```diff
@@ -75,18 +73,57 @@ class HL7DiffHelper {
                     }
                     if (outputFields.size > inputFields.size) {
```

Comment:
Would it be okay to move the effectivelyBlank call earlier and then take it out of the `foldIndexed` lambdas? It might make this PR a lot simpler, and also would short-circuit a lot of the logic.
```suggestion
                    }
                    if (effectivelyBlank(outputFields, inputFields)) {
                        continue
                    }
                    if (outputFields.size > inputFields.size) {
```